src/react/components/common/cloudFilePicker/cloudFilePicker.tsx

```diff
@@ -133,13 +169,58 @@ export class CloudFilePicker extends React.Component<ICloudFilePickerProps, IClo
             const storageProvider = StorageProviderFactory.createFromConnection(this.state.selectedConnection);
             const content = await storageProvider.readText(this.state.selectedFile);
             this.props.onSubmit(content);
+        } else if (this.state.pastedUri && this.getSharedProjectConnectionInfo()) {
```

Comment:
Instead of call getSharedProjectConnectionInfo() twice, should assign result as variable